### PR TITLE
Add Basic Auth header option

### LINKED ISSUE #41: Add option for basic auth that sets header with token
https://en.wikipedia.org/wiki/Basic_access_authentication#Client_side

### PR DESCRIPTION
**Pull request by rkillackey**
Resolves #41 

Assuming that the user is passing `username:password` as `bearerToken`

### CODE REVIEW

**Comment by rkillackey** [comment]
@dpikt Ready for review!

File: src/http/middleware/set-auth-headers.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
-function addAuthHeaders ({ bearerToken, headers, overrideHeaders=false }) {
-  if (overrideHeaders || !bearerToken) return
+function addAuthHeaders ({ 
+  auth,
+  bearerToken, 
+  headers, 
+  overrideHeaders=false, 
+}) {
+  if (overrideHeaders) return
+  let authToken = bearerToken
```
Comment (dpikt) [inline src/http/middleware/set-auth-headers.js]: Eek, a`let`!! 🙅‍♂️ 😅 

Don't be afraid to have two return statements, even if it's a bit redundant:
```
const basicAuthToken = encodeBasicAuth(auth)
if (basicAuthToken) return { ... }
if (bearerToken) return { ... }
```

File: src/http/middleware/set-auth-headers.js [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+  if (auth) {
+    const username = auth.username || ''
+    const password = auth.password || ''
+    authToken = window.btoa(`${ username }:${ password }`)
```
Comment (dpikt) [inline src/http/middleware/set-auth-headers.js]: Use an [isomorphic](https://www.npmjs.com/package/Base64) version of this to support node as well

File: src/http/middleware/set-auth-headers.js (hunk 1 above)
Comment (dpikt) [reply src/http/middleware/set-auth-headers.js]: Alternatively with one:
```
const basicAuthToken = encodeBasicAuth(auth)
if (!(basicAuthToken || bearerToken)) return
const authHeader = basicAuthToken ? ...
return { ... }
```

**Pull request merged**